plugins/japanese.koplugin/deinflector.lua

```diff
@@ -0,0 +1,470 @@
+    local item_table = {}
+    for _, conversion in pairs(ALL_TEXT_CONVERSIONS) do
+        local name = conversion.name
+        table.insert(item_table, {
+            text = conversion.pretty_name,
+            help_text = conversion.help_text,
+            checked_func = function()
+                return self.enabled_text_conversions[name] or false
+            end,
+            callback = function(touchmenu_instance)
+                self.enabled_text_conversions[name] = not self.enabled_text_conversions[name]
+                G_reader_settings:saveSetting("language_japanese_text_conversions", self.enabled_text_conversions)
+                if touchmenu_instance then touchmenu_instance:updateItems() end
+            end,
+        })
+    end
+    return item_table
+end
+
+function Deinflector:genMenuItems()
+    return {
+        {
+            text_func = function()
+                local nenabled = 0
+                for _, enabled in pairs(self.enabled_text_conversions) do
+                    if enabled then nenabled = nenabled + 1 end
+                end
+                if nenabled == 0 then
+                    return _("Text conversions: none enabled")
+                else
+                    return T(N_("Text conversions: %1 enabled", "Text conversions: %1 enabled", nenabled), nenabled)
+                end
+            end,
+            help_text = _([[
+Configure which text conversions to apply when trying to deinflect Japanese text. These primarily include conversions between different kinds of kana, in order to make sure that a word written using different kana to your installed dictionaries can still be looked up.
+
+Not every conversion will be applied at once. Instead, all possible combinations of enabled conversions will be attempted in order to maximise the chance of at least one conversion matching the form used in the dictionary.]]),
+            sub_item_table = self:genTextConversionMenuItems(),
+        },
+        {
+            -- @translators A deinflector is a program which converts a word into its dictionary form, similar to deconjugation in European languages. See <https://en.wikipedia.org/wiki/Japanese_verb_conjugation> for more detail.
+            text = _("Deinflector information"),
+            keep_menu_open = true,
+            callback = function()
+                local nrules, nvariants = 0, 0
+                for _, rules in pairs(self.rules) do
+                    nvariants = nvariants + #rules
+                    nrules = nrules + 1
+                end
+                local nrules_str = T(N_("%1 rule", "%1 rules", nrules), nrules)
+                local nvariants_str = T(N_("%1 variant", "%1 variants", nvariants), nvariants)
+                UIManager:show(InfoMessage:new{
+                    -- @translators %1 is the "%1 rule(s)" string, %2 is the "%1 variant(s)" string.
+                    text = T(_("Deinflector has %1 and %2 loaded."), nrules_str, nvariants_str),
```

Comment:
@cyphar There are some complaints about this construction.

![image](https://user-images.githubusercontent.com/202757/222892825-674a83a8-ca26-479c-8e18-0c163f5677ea.png)

<https://hosted.weblate.org/translate/koreader/koreader/en/?checksum=66e64e8b6763628e>

Reply:
The issue is that there are two numerical values in the string which need to be pluralised, and (at least when I checked while writing this) you cannot have more than one variables in a single string with pluralisation. While the values are never going to be 0 or 1, some languages have complicated pluralising rules so just combining them into one string would also make translations difficult or impossible.

The only way I can see to make this easier to translate is to make it two sentences instead (though it would make the info screen more wordy)?

Reply:
Yes, without having two separate sentences it can be done in GetText, only in Fluent or some such.

Or alternatively something like

Deinflector loaded:
 * 1 rule(s)
 * 1 variant(s)